File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03665/p03665_num7_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, P;
    cin >> N >> P;

    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    long long count = 0;
    for (int i = 0; i < (1 << N); i++) {
        int sum = 0;
        for (int j = 0; j < N; j++) {
            if ((i >> j) & 1) {
                sum += A[j];
            }
        }
        if (sum % 2 == P) {
            count++;
        }
    }

    cout << count << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03665/p03665_num7_parsed.py
# Read the input values
N, P = map(int, input().split())
A = list(map(int, input().split()))

# Generate all possible subsets
subsets = []
for i in range(2**N):
    subset = []
    for j in range(N):
        if (i >> j) & 1:
            subset.append(A[j])
    subsets.append(subset)

# Count the number of subsets with a total sum congruent to P modulo 2
count = 0
for subset in subsets:
    if sum(subset) % 2 == P:
        count += 1

# Print the result
print(count)